docs/advanced/persistence-and-collaboration.mdx

```diff
@@ -0,0 +1,115 @@
+---
+title: Persistence and Collaboration
+---
+
+## Introduction
+
+By default `Variable` state is purely client-side - by default its source of truth is the browser memory.
+It also means that refreshing the browser page resets the `Variable` state, and that each user has their own 'copy'
+of the data. To customize the behaviour, `Variable` accepts a store property. This allow changing the `Variable`'s source of truth,
+thereby enabling features like data persistence and collaborative interactions across users.
+
+The rest of the page goes in-depth into available stores and how they can be used in your application.
+
+
+## `dara.core.persistence.BackendStore`
+
+This type of store changes the source of truth for a given `Variable` to live on the server, rather than on the client.
+This has two main benefits:
+- the data can be persistent across page refreshes or even across server restarts
+- since the data lives server-side, we can 'sync' it across clients to enable collaboration
+
+To use a `BackendStore`, simply create a store instance and attach it to a `Variable`:
+
+```python
+from dara.core import Variable
+from dara.core.persistence import BackendStore
+from dara.components import Input
+
+# Tag the variable to be 'collaborative' and have a shared server state
+collab_variable = Variable(default=1, store=BackendStore())
+
+# Input value will automatically be kept in sync 'live' for all users
+Input(value=collab_variable)
+```
+
+Under the hood the `BackendStore` mechanism is very simple:
+- the store data is initialized with the `default` `Variable` value
+- on initial page load, the `Variable` value is fetched from the server
+- whenever a client makes an update, it makes a request to the server to update the state,
+  which then notifies all other clients about the new state
+
+You can interact with the store programmatically to read, write or delete data from the store.
+
+```python
+from dara.core import Variable
+from dara.core.persistence import BackendStore
+from dara.components import Input
+
+store = BackendStore()
+collab_variable = Variable(default='default value', store=store)
+
+# get current value in the store
+value = await store.read()
+# store can also be accessed from the variable
+value = await variable.store.read()
```

Comment:
I am guessing these print the same thing?

Reply:
Yeah that's the idea, I'm just trying to show that you can access it from the variable so you don't need to keep around a reference to the store itself.